src/Crate/PDO/Http/Client.php

```diff
@@ -61,29 +55,31 @@ public function execute($query, array $parameters)
             'args' => $parameters
         ];
 
-        $request = new Request();
-        $request->setUri($this->uri);
-        $request->setMethod('POST');
-        $request->setBody(json_encode($body));
+        try {
+            $response     = $this->client->post(null, ['json' => $body]);
+            $responseBody = $response->json();
+
+            return new Collection(
+                $responseBody['rows'],
+                $responseBody['cols'],
+                $responseBody['duration'],
+                $responseBody['rowcount']
+            );
 
-        $promise     = $this->client->request($request);
-        $response    = $promise->wait();
-        $responseBody = json_decode($response->getBody(), true);
+        } catch (ClientException $exception) {
 
-        if ($response->getStatus() !== 200) {
+            $json = $exception->getResponse()->json();
 
-            $errorCode    = $responseBody['error']['code'];
-            $errorMessage = $responseBody['error']['message'];
+            $errorCode    = $json['error']['code'];
+            $errorMessage = $json['error']['message'];
 
             throw new RuntimeException($errorMessage, $errorCode);
-        }
 
-        return new Collection(
-            $responseBody['rows'],
-            $responseBody['cols'],
-            $responseBody['duration'],
-            $responseBody['rowcount']
-        );
+        } catch (ServerException $exception) {
+
+            // todo: this could/should probably be handled here as well
```

Comment:
@macnibblet @Ocramius crate respond with a 5xx if an unexpected error happens like e.g. a shard failure. so i think re-throwing the exception would be ok. (sorry for the delay, was on holiday ;)